### PR TITLE
Feature/refresh token functionality

### PR DESCRIPTION
**Pull request by VitaliiDenysiuk**

### CODE REVIEW

File: src/config/jwt.ts [hunk 1]
```diff
@@ -1,25 +1,69 @@
...
+  const findTokenInDatabase = await getToken(refreshToken);
+  if (!findTokenInDatabase) {
+    throw Error("Invalid refresh token");
+  }
+  const verifyPeyload: User = verify(
+    refreshToken,
+    process.env.JWT_SECRET,
+    verifyOptions
+  ) as User;
+  if (verifyPeyload._id !== findTokenInDatabase.userId) {
```
Comment (AndrewKaenko) [inline src/config/jwt.ts]: Spelling mistake - peyload -> payload

File: src/middlewares/isAuthorised.middlewares.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  message: "Access denied. No token provided.",
+  status: 403,
+};
+
+export const isAuthorised = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  // try {
```
Comment (AndrewKaenko) [inline src/middlewares/isAuthorised.middlewares.ts]: Why this detail?

**Review by AndrewKaenko — APPROVED** [review]

File: src/config/jwt.ts (hunk 1 above)
Comment (VitaliiDenysiuk) [reply src/config/jwt.ts]: corrected

File: src/middlewares/isAuthorised.middlewares.ts (hunk 2 above)
Comment (VitaliiDenysiuk) [reply src/middlewares/isAuthorised.middlewares.ts]: remove

**Review by fominmaksim — APPROVED** [review]
lgtm

File: src/middlewares/isAuthorised.middlewares.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { NextFunction, Response, Request } from "express";
```
Comment (alexandrtovmach) [inline src/middlewares/isAuthorised.middlewares.ts]: isAuthorized

File: src/middlewares/isAuthorised.middlewares.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import { verifyJWT } from "../config/jwt";
+import CustomError from "../models/CustomError.model";
+import { User } from "../models/User.model";
+
+const accessDenied: CustomError = {
+  name: "AccessDenied",
+  message: "Access denied. No token provided.",
+  status: 403,
```
Comment (alexandrtovmach) [inline src/middlewares/isAuthorised.middlewares.ts]:
```suggestion
  status: 401,
```

File: src/routes/auth.route.ts [hunk 5]
```diff
@@ -29,4 +29,13 @@ authRouter.route("/login").post(async (req: Request, res: Response) => {
   res.send(result);
 });
 
+authRouter.route("/refreshToken").post(async (req: Request, res: Response) => {
```
Comment (alexandrtovmach) [inline src/routes/auth.route.ts]:
```suggestion
authRouter.route("/refresh").post(async (req: Request, res: Response) => {
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**